Quake/menu.c

```diff
@@ -1157,20 +1185,37 @@ void M_Menu_Options_f (void)
 	m_entersound = true;
 }
 
-void M_AdjustSliders (int dir)
+void M_AdjustSliders (int dir, qboolean mouse)
 {
 	int	  curr_alwaysrun, target_alwaysrun;
-	float f, l;
+	float f, l, clamped_mouse = CLAMP (SLIDER_START, (float)m_mouse_x, SLIDER_END);
+
+	if (fabsf (clamped_mouse - (float)m_mouse_x) > 12.0f)
+		mouse = false;
 
 	S_LocalSound ("misc/menu3.wav");
 
 	switch (options_cursor)
 	{
+	case OPT_FOV:
+		if (mouse)
+			f = 80 + (130 - 80) * (clamped_mouse - SLIDER_START) / SLIDER_EXTENT;
+		else
+			f = scr_fov.value + dir * 5;
+		if (f < 80)
+			f = 80;
+		else if (f > 130)
+			f = 130;
+		Cvar_SetValue ("fov", f);
+		break;
 	case OPT_SCALE: // console and menu scale
 		if (scr_relativescale.value)
 		{
 			l = 3.0f;
-			f = scr_relativescale.value + dir * .1;
+			if (mouse)
```

Comment:
I think we should add a function for those to remove the repetition of this if/else pattern?

Reply:
Yeah, it should be better. I've factored it out now.